src/AppInstallerCLICore/Workflows/GetRequestedUrlFromManifest.cpp

```diff
@@ -0,0 +1,71 @@
+#include "pch.h"
+#include "GetRequestedUrlFromManifest.h"
+
+#include "ManifestComparator.h"
+#include "ShellExecuteInstallerHandler.h"
+
+using namespace AppInstaller::CLI;
+using namespace AppInstaller::Utility;
+using namespace AppInstaller::Manifest;
+
+namespace AppInstaller::CLI::Workflow
+{
+	void GetRequestedUrlFromManifest(Execution::Context& context)
```

Comment:
>void GetRequestedUrlFromManifest(Execution::Context& context) [](start = 1, length = 61)

[Get ready for my manifesto on the workflow and context object.]

The goal of the workflow design was to prevent the need to have these kinds of nested if statements.  Each workflow step can be seen as a transform (or transition in the state machine) that takes the previous state and produces a new one.  Any transform also has the option of failing the workflow, and thus future steps need not worry about the presence or correctness of the data (assuming that the developer properly constructed the flow).

So in the case of this function, we need to extract a string from the manifest.  Assuming that we have put this after the appropriate prior transforms (since show requires them, you have), we can be assured that there is already a single manifest stored in the context and can extract it directly.  So you should be able to start at line 24 (```const auto& manifest = context.Get<Execution::Data::Manifest>();```) and remove the outer scopes.

As for the context object, the data it stores is intended to be how information is passed from one transform to another.  This can be either because it is expensive to acquire, or to allow divergent paths to converge.  I don't see that this use case needs either; you can simply extract the appropriate string, then use it later in the same function.  If we have need to launch arbitrary URLs later, we can refactor the code back out.  In either case, a single Data entry should be used (ex. Data::LaunchURL) so that it can be generically applied.

Reply:
Just got to this point, I've removed the outer scopes and as i read and understand you correctly your saying extract and make available one url. then in a later workflow use it. 
what I've done with this part of the code is allow you to call 
```
winget show git --homepage --license
```
which would launch BOTH url's 
I could convert this "Data::LaunchUrls" as a string array of some sort. and process that. 
OR for the matter of simplification and expectation remove the duality in the command. 
which would be advisable? 
1.  Data::LaunchUrls
2. Launch only one URL at a time per command.
I think the proper way is # 2 but I like the thought of having all the urls specified and available open as requested in one command.

Reply:
I wouldn't even really bother with putting it into the context, just extract and use it right there.  But if you did, then yes, collapse it to one data member (either a single string or a vector of them).

I don't mind whether it is one or multiple honestly; the user is in control of requesting it, and they won't really interfere with each other.

Reply:
I created a vector of string, and changed the Open method to just go over this list and execute. 
the concept is that additional methods can add to this list in the future and can open any number of urls.
Thanks for the hint of the vector :) 
the addition of the url's to the vector handles all the logic for parsing which ones to add so i removed the logic to parse which ones to show.